.dagger/sdk_php.go

```diff
@@ -6,138 +6,123 @@ import (
-		_, err := dag.PhpSDKDev(dagger.PhpSDKDevOpts{Source: src}).Lint().Sync(ctx)
-		return err
-	})
-
-	eg.Go(func() (rerr error) {
-		ctx, span := Tracer().Start(ctx, "check that the generated client library is up-to-date")
-		defer func() {
-			if rerr != nil {
-				span.SetStatus(codes.Error, rerr.Error())
-			}
-			span.End()
-		}()
-		before := t.Dagger.Source
-		after, err := t.Generate(ctx)
-		if err != nil {
-			return err
-		}
-		return dag.Dirdiff().AssertEqual(ctx, before, after, []string{
-			filepath.Join(phpSDKPath, phpSDKGeneratedDir),
-			phpSDKGeneratedDocsDir,
-		})
-	})
+func (t PHPSDK) Name() string {
+	return "php"
+}
 
-	return eg.Wait()
+func (t PHPSDK) Source() *dagger.Directory {
+	return t.Dagger.Source.Directory("sdk/php")
+}
+
+// Lint the PHP SDK
+func (t PHPSDK) CheckLint(ctx context.Context) error {
+	_, err := dag.PhpSDKDev(dagger.PhpSDKDevOpts{Source: t.Source()}).
+		Lint().
+		Sync(ctx)
+	return err
 }
 
 // Test the PHP SDK
 func (t PHPSDK) Test(ctx context.Context) error {
-	installer := t.Dagger.installer("sdk")
-	src := t.Dagger.Source.Directory(phpSDKPath)
 	base := dag.PhpSDKDev().Base().
-		With(installer).
+		With(t.Dagger.devEngineSidecar()).
 		WithEnvVariable("PATH", "./vendor/bin:$PATH", dagger.ContainerWithEnvVariableOpts{Expand: true})
 
-	dev := dag.PhpSDKDev(dagger.PhpSDKDevOpts{Container: base, Source: src})
+	dev := dag.PhpSDKDev(dagger.PhpSDKDevOpts{Container: base, Source: t.Source()})
 	_, err := dev.Test().Sync(ctx)
 	return err
 }
 
 // Regenerate the PHP SDK API + docs
-func (t PHPSDK) Generate(ctx context.Context) (*dagger.Directory, error) {
-	src := t.Dagger.Source.Directory(phpSDKPath)
-
-	api := t.generateAPI(src)
-
-	// NB: we need to chain the API generation into the docs generation, since
-	// doctum analyzes the php code to generate docs
-	docs, err := t.generateDocs(ctx, src.WithDirectory("", api))
+func (t PHPSDK) Generate(ctx context.Context) (*dagger.Changeset, error) {
+	genClient := t.generateClient()
+	genDocs, err := t.generateDocs(ctx, genClient)
 	if err != nil {
 		return nil, err
 	}
-
-	return dag.Directory().
-		WithDirectory("", api).
-		WithDirectory("", docs), nil
+	src := t.Dagger.Source
+	return src.
+		WithChanges(genClient).
+		WithChanges(genDocs).
+		Changes(src).
+		Sync(ctx)
 }
 
-func (t PHPSDK) generateAPI(src *dagger.Directory) *dagger.Directory {
-	installer := t.Dagger.installer("sdk")
-	generated := dag.PhpSDKDev(dagger.PhpSDKDevOpts{Source: src}).
+func (t PHPSDK) generateClient() *dagger.Changeset {
+	src := t.Source()
+	relLayer := dag.PhpSDKDev(dagger.PhpSDKDevOpts{Source: src}).
 		Base().
-		With(installer).
-		WithoutDirectory(phpSDKGeneratedDir).
-		WithDirectory(phpSDKGeneratedDir, dag.Directory()).
+		With(t.Dagger.devEngineSidecar()).
+		WithoutDirectory("generated").
+		WithDirectory("generated", dag.Directory()).
+		// FIXME: why not inject the right dagger binary, instead of leaking this env var?
```

Comment:
Mmmmm. There's some magic in our `core/integration` tests that use this, but for our use here, we don't need to be setting it.

The original source code for `devEngineSidecar` was pretty much copy-pasted from `core/integration` so that's why it would have it in the first place.